File: HeraldTest-Android/app/src/main/java/com/example/heraldtest/PeerInfo.java
package com.example.heraldtest;

import java.util.ArrayList;
import io.heraldprox.herald.sensor.datatype.Date;

// https://github.com/broadinstitute/operation-outbreak/issues/163
public class PeerInfo {
    ArrayList<Double> data;
    IllnessStatus status;
    Date lastSeen;

    PeerInfo() {
        this.data = new ArrayList<>();
        this.status = new IllnessStatus(IllnessStatusCode.susceptible, new Date());
        this.lastSeen = new Date();
    }

    void setStatus(IllnessStatus status) {
        this.status = status;
        this.lastSeen = new Date();
    }

    void addRSSI(double value) {
        if (-100 <= value && value < 0) {
            data.add(value);
            if (25 < data.size()) {
               data.remove(0);
            }
        }

    }

    double getRSSI() {
        if (0 < data.size()) {
            double sum = 0;
            for (Double v: data) sum += v;
            return sum / data.size();
        } else {
            return -100;
        }
    }
}

File: HeraldTest-iOS/HeraldTest/PeerInfo.swift
//
//  PeerInfo.swift
//  HeraldTest
//
//

import Foundation

class PeerInfo {
    public var data: [Double] = []
    public var status: IllnessStatus
    public var lastSeen: Date
    
    init(data: [Double], status: IllnessStatus, lastSeen: Date) {
        self.data = data
        self.status = status
        self.lastSeen = lastSeen
    }
    
    init() {
        self.data = []
        self.status = IllnessStatus.init(status: .susceptible, dateSince: Date())
        self.lastSeen = Date()
    }
    
    func setStatus(_ status: IllnessStatus) {
        self.status = status
        self.lastSeen = Date()
    }
    
    func addRSSI(_ value: Double) {
        if (-100 <= value && value < 0) {
            data.append(value);
            if (25 < data.count) {
               data.remove(at:0);
            }
        }
    }
    
    func getRSSI() -> Double {
        if (0 < data.count) {
            var sum = 0.0;
            for v in data {
                sum += v
            }
            return sum / Double(data.count);
        } else {
            return -100;
        }
    }
}
